SU2_CFD/include/solvers/CNEMOEulerSolver.hpp

```diff
@@ -199,6 +214,12 @@ class CNEMOEulerSolver : public CFVMFlowSolverBase<CNEMOEulerVariable, COMPRESSI
    */
   void SetNondimensionalization(CConfig *config, unsigned short iMesh) final;
 
+    /*!
+   * \brief Compute the pressure at the infinity.
+   * \return Value of the pressure at the infinity.
+   */
+  inline CNEMOGas* GetFluidModel(void) const final { return FluidModel;}
+
```

Comment:
You need to be a little careful with this because if you call this method on a CSolver pointer to a CNEMOEulerSolver instance, the return is a pointer to CFluidModel and not to CNEMOGas, which may or may not be ok.
```c++
CNEMOEulerSolver* asolver = new CNEMOEulerSolver(...);
asolver->GetFluidModel(); // <- This is a CNEMOGas*

CSolver* bsolver = asolver;
bsolver->GetFluidModel(); // <-  But this is a CFluidModel*
```

Reply:
I can go through and change this.  This may cause problems in the turbulence solver eventually.

Reply:
@pcarruscag any update?

Reply:
can.'t we simply send the pointer FluidModel we use throughout the CNEMOEUlerSolver file ?

Reply:
Ah sorry, I did not see Wally's message was edited (github does not notify for that).

You can access the fluid model variable directly within the solvers, or leave the code as is.
You can worry about the turbulence when you get there, a CFluidModel can be forced to CNEMOGas with a static cast, IMO that is relatively safe thing to in places where it is known that pointer can only refer to a CNEMOGas.

Reply:
ok so it seems that, for now, "access the fluid model variable directly within the solvers" is the simplest solution, I think this is the same that I suggested, correct ? just to double-check

Reply:
Sorry, I need to be a bit better about my editting....

In any case, when trying to directly access the FluidModel, there is an issue regarding static members.  I am wondering since this function is a static member, I do need to pass the fluid model in?